src/mesi_search/main.py

```diff
@@ -58,7 +47,17 @@ def discover_candig_patient():
     private_filtered_data = candig.private_data_filter(data=raw_results,
                                                        terms=attribute_of_interest,
                                                        path="/results/patients")
+    # save attributes of interest to session, as well as result this is to save
+    # recalculation and send different data for the same user.
+    # TODO: Save the session in persistent storage
+    attrs_of_interest_from_session = session.get("attributes_of_interest", [])
+    attrs_of_interest_from_session += attribute_of_interest
+    attrs_of_interest_from_session = list(set(attrs_of_interest_from_session))
+    session["attributes_of_interest"] = attrs_of_interest_from_session
```

Comment:
These 4 lines of code could be converted to the following. IMO this is slightly more readable as you're deduplicating `attrs_of_interest_from_session` from the beginning instead of doing it at the end by converting it to a set and back to a list.
```
attrs_of_interest_from_session = set(session.get("attributes_of_interest", []))
attrs_of_interest_from_session.update(attribute_of_interest)
session["attributes_of_interest"] = list(attrs_of_interest_from_session)
```

Reply:
Yea, thank you for this. I felt uneasy with that code and I am still not happy but yours is much better and a vast improvement!